### PR TITLE
Pause doc: World map points

### PR DESCRIPTION
**Pull request by Dragorn421**
- the `QUAD_MAP_WORLD_CLOUDS_` part of the `MapQuad` enum is derived from the `cloudTexs`, `cloudFlagNums` arrays in `KaleidoScope_DrawWorldMap`
- the `WorldMapPoint` enum is derived from `g*PointNameENGTex` textures in `map_name_static`.
  The array of textures in `map_name_static` is indexed in `KaleidoScope_UpdateNamePanel` by `texIndex` aka `pauseCtx->cursorItem[PAUSE_MAP]`, and `cursorItem[PAUSE_MAP]` is set to `WorldMapPoint` values in `KaleidoScope_DrawWorldMap`: `pauseCtx->cursorItem[PAUSE_MAP] = pauseCtx->cursorPoint[PAUSE_WORLD_MAP];`

### CODE REVIEW

File: src/overlays/misc/ovl_kaleido_scope/z_kaleido_map.c [hunk 1]
```diff
@@ -720,7 +735,7 @@ void KaleidoScope_DrawWorldMap(PlayState* play, GraphicsContext* gfxCtx) {
         gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 235, 235, 235, pauseCtx->alpha);
 
         for (k = 0; k < 15; k += 8) {
-            gSPVertex(POLY_OPA_DISP++, &pauseCtx->mapPageVtx[60 + k * 4], 32, 0);
+            gSPVertex(POLY_OPA_DISP++, &pauseCtx->mapPageVtx[(PAGE_BG_QUADS + k) * 4], 32, 0);
```
Comment (cadmic) [inline src/overlays/misc/ovl_kaleido_scope/z_kaleido_map.c:738]: how is adding parens here equivalent?

Comment (Dragorn421) [reply src/overlays/misc/ovl_kaleido_scope/z_kaleido_map.c:738]: `PAGE_BG_QUADS` expands to `3*5` = 15, and `15*4=60`

Comment (cadmic) [reply src/overlays/misc/ovl_kaleido_scope/z_kaleido_map.c:738]: right but before it's 60 + (k * 4) and now it's (60 + k) * 4

File: include/z64pause.h [hunk 2]
```diff
@@ -96,6 +96,28 @@ typedef enum PauseMainState {
     /* 9 */ PAUSE_MAIN_STATE_9
 } PauseMainState;
 
+typedef enum WorldMapPoint {
+    /*  0 */ WORLD_MAP_POINT_HAUNTED_WASTELAND,
+    /*  1 */ WORLD_MAP_POINT_GERUDOS_FORTRESS,
+    /*  2 */ WORLD_MAP_POINT_GERUDO_VALLEY,
+    /*  3 */ WORLD_MAP_POINT_HYLIA_LAKESIDE,
```
[the remark this answers is not in the document]
Comment (Dragorn421) [reply include/z64pause.h:103]: interesting, TIL the `gHyliaLakesidePointNameENGTex` texture spells out "Lake Hylia" in gc-us/gc-jp and "Hylika Lakeside" in gc-eu
(these docs were made with gc-eu-mq-dbg)

I'll change to LAKE_HYLIA

File: src/overlays/misc/ovl_kaleido_scope/z_kaleido_map.c (hunk 1 above)
Comment (cadmic) [reply src/overlays/misc/ovl_kaleido_scope/z_kaleido_map.c:738]: (perhaps k * 4 + 60 or 4 * k + 60 would be better if it matches)

Comment (Dragorn421) [reply src/overlays/misc/ovl_kaleido_scope/z_kaleido_map.c:738]: before:
`60 + k * 4`
after:
`(PAGE_BG_QUADS + k) * 4`
aka `(15 + k) * 4`, is equivalent

Comment (cadmic) [reply src/overlays/misc/ovl_kaleido_scope/z_kaleido_map.c:738]: oh... sorry

**Review by cadmic — APPROVED** [review]

File: src/overlays/misc/ovl_kaleido_scope/z_kaleido_scope.h [hunk 3]
```diff
@@ -40,6 +40,36 @@ extern u8 gAreaGsFlags[];
 #define WORLD_MAP_IMAGE_FRAG_HEIGHT ((TMEM_SIZE / 2) / (WORLD_MAP_IMAGE_WIDTH * G_IM_SIZ_8b_BYTES))
 #define WORLD_MAP_IMAGE_FRAG_NUM (((WORLD_MAP_IMAGE_HEIGHT - 1) / WORLD_MAP_IMAGE_FRAG_HEIGHT) + 1)
 
+// Clouds quads cover undiscovered areas.
```
Comment (fig02) [inline src/overlays/misc/ovl_kaleido_scope/z_kaleido_scope.h:43]: double plural sounds awkward "cloud quads" seems fine

**Review by fig02 — APPROVED** [review]

**Comment by fig02** [comment]
can merge after conflict resolved

**Pull request merged**